### PR TITLE
JAMES-2586 Implement PostgresEventDeadLetters

### PR DESCRIPTION
**Pull request by quantranhong1999**
resolve https://github.com/linagora/james-project/issues/4985

### CODE REVIEW

**Comment by quantranhong1999** [comment]
We have an unstable test: `guiceServerShouldUpdateQuota(GuiceJamesServer) – org.apache.james.DistributedPostgresJamesServerTest`. (reported by @vttranlina a while ago anyway)

I will try to quickly tackle it tmr. Otherwise, I will disable it and create a dedicated ticket.

**Comment by vttranlina** [comment]
maybe listener `ListeningCurrentQuotaUpdater` needs more time to complete (after implementing event bus with rabbitmq)

File: event-bus/postgres/pom.xml [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Arsnael) [inline event-bus/postgres/pom.xml:1]: Missing license?

File: event-bus/postgres/src/main/java/org/apache/james/events/PostgresEventDeadLettersModule.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import org.jooq.Table;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+
+public interface PostgresEventDeadLettersModule {
+    interface PostgresEventDeadLettersTable {
+        Table<Record> TABLE_NAME = DSL.table("event_dead_letters");
+
+        Field<UUID> INSERTION_ID = DSL.field("insertion_id", SQLDataType.UUID.notNull());
+        Field<String> GROUP = DSL.field("\"group\"", SQLDataType.VARCHAR.notNull());
```
Comment (Arsnael) [inline event-bus/postgres/src/main/java/org/apache/james/events/PostgresEventDeadLettersModule.java:38]: Any particular reason you need to add those `\"` in your syntax here?

Comment (quantranhong1999) [reply event-bus/postgres/src/main/java/org/apache/james/events/PostgresEventDeadLettersModule.java:38]: `group` is a Postgres keyword so I chose to escape it as a normal identifier.

Comment (hungphan227) [inline event-bus/postgres/src/main/java/org/apache/james/events/PostgresEventDeadLettersModule.java:38]: Another option is changing column name to event_group or sth like that

Comment (quantranhong1999) [reply event-bus/postgres/src/main/java/org/apache/james/events/PostgresEventDeadLettersModule.java:38]: Yep, I knew, but I chose to keep `group`.

**Review by hungphan227 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

File: event-bus/postgres/src/main/java/org/apache/james/events/PostgresEventDeadLettersModule.java (hunk 2 above)
Comment (Arsnael) [reply event-bus/postgres/src/main/java/org/apache/james/events/PostgresEventDeadLettersModule.java:38]: I'm ok with this. let's keep it consistent with our other implems

**Review by vttranlina — APPROVED** [review]

**Pull request merged**